Watchman.Discord/Areas/Protection/Strategies/LinksDetectorStrategy.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Text.RegularExpressions;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam.Models;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+
+namespace Watchman.Discord.Areas.Protection.Strategies
+{
+    public class LinksDetectorStrategy : ISpamDetector
+    {
+        public IUserMessagesCounter UserMessagesCounter { get; set; }
+
+        private readonly int _userMessagesCountToBeSafe;
+        
+        public LinksDetectorStrategy(IUserMessagesCounter userMessagesCounter)
+        {
+            UserMessagesCounter = userMessagesCounter;
+            _userMessagesCountToBeSafe = 500;
+        }
+
+        public SpamProbability GetSpamProbability(ServerMessagesCacheService serverMessagesCacheService, DiscordRequest request, Contexts contexts)
+        {
+            var content = request.OriginalMessage;
+            var linkRegex = new Regex(@"http[s]?:\/\/");
```

Comment:
czemu s w []?

Reply:
https://stackoverflow.com/questions/3809401/what-is-a-good-regular-expression-to-match-a-url